addons/mendeley/models.py

```diff
@@ -267,3 +269,32 @@ class NodeSettings(BaseCitationsNodeSettings):
     def _fetch_folder_name(self):
         folder = self.api._folder_metadata(self.list_id)
         return folder.name
+
+    def get_folders(self, show_root=False, **kwargs):
+        if self.has_auth:
+            try:
+                folders = self.api._get_folders()
+                serialized_root_folder = {
+                    'name': 'All Documents',
+                    'provider_list_id': None,
+                    'id': 'ROOT',
+                    'parent_list_id': '__',
+                    'kind': 'folder',
+                    'addon': 'mendeley'
+                }
+                serialized_folders = [{
+                    'addon': 'mendeley',
+                    'kind': 'folder',
+                    'id': folder.json['id'],
+                    'name': folder.json['name'],
+                    'path': folder.json.get('parent_id', '/'),
+                    'parent_list_id': folder.json.get('parent_id', None),
+                    'provider_list_id': folder.json['id']
+                } for folder in folders]
+                if show_root:
+                    serialized_folders.insert(0, serialized_root_folder)
+                return serialized_folders
+            except MendeleyApiException:
+                return []
```

Comment:
Drive-by: Looks like we handle 3rd-party API exceptions differently across addons. Of the ones that have been refactored for v2:

* Google Drive: No error handling
* Box: reraises HTTPError
* Dropbox: reraises HTTPError
* S3: raraises InvalidAuthError

Furthermore, it doesn't appear that we're handling the re-raised errors in v2:

https://github.com/CenterForOpenScience/osf.io/blob/1979ed6a2d19989bdb502b33553513f6678c78f8/api/nodes/views.py#L1152 

The most correct approach is probably to reraise an HTTPError when we get an error response from Mendeley (unless Mendeley does something weird like return an error response for empty folders or something like that).

Then we should make a ticket to handle HTTPErrors in the APIv2 view.

Reply:
Made a follow-up ticket to handle the v1 HTTPErrors: https://openscience.atlassian.net/browse/OSF-9226